mvpa2/base/dataset.py

```diff
@@ -795,7 +795,18 @@ def all_equal(x, y):
     try:
         eq = x == y
     except:
-        return False
+        # we can get here, in case of dictionaries with non-simple values
+        # (e.g. arrays)
+        try:
+            if not set(x.keys()) == set(y.keys()):
+                return False
+            x = [x[k] for k in x]
+            y = [y[k] for k in y]
```

Comment:
It seems that "k in x" can return keys in arbitrary order, so maybe better to do:

keys=list(x.keys()); % by python3 compatible
x = [x[k] for k in keys]
y = [y[k] for k in keys]